PlaceRouteHierFlow/placer/ILP_solver.cpp

```diff
@@ -3462,8 +3462,27 @@ double ILP_solver::GenerateValidSolution(const design& mydesign, const SeqPair&
       }
     }
     for (unsigned i = 0; i < mydesign.Blocks.size(); i++) {
-      roundup(Blocks[i].x, x_pitch);
-      roundup(Blocks[i].y, y_pitch);
+      bool non_zero_xoffset = false, non_zero_yoffset = false;
+      for (auto instance : mydesign.Blocks[i]) {
+        for (auto offset : instance.xoffset) {
+          if (offset != 0) {
+            non_zero_xoffset = true;
+            break;
+          }
+        }
+        if (non_zero_xoffset) break;
+      }
+      if (!non_zero_xoffset) roundup(Blocks[i].x, x_pitch);
```

Comment:
The idea here is: if a non zero offset exists, place_on_grid has priority over poly pitch. Is this okay?

Reply:
This might be fine. Yet could this result in an overlap? Suppose A and B blocks placed next to each other AB and B has offset. This means B is not round up but A might be round up and overlap with B?

Reply:
this may happen

Reply:
Could this legalization be avoided if all blocks have a PlaceOnGrid,V constraint? Of course, this would have a significant impact on ILP runtime.

Reply:
I suggest we merge this PR and let user floorplan to prevent this from happening.

Reply:
Yes. if all blocks have place_on_grid, then they they are not rounded up.